addons/stock/migrations/8.0.1.1/post-migration.py

```diff
@@ -0,0 +1,714 @@
+           LEFT JOIN product_product p ON o.product_id = p.id
+           LEFT JOIN product_template t ON p.product_tmpl_id = t.id
+           WHERE t.uom_id != o.%(product_uom)s
+        ''' % {'product_uom': openupgrade.get_legacy_name('product_uom')}
+    cr.execute(Q)
+    if cr.rowcount == 0:
+        return
+    logger.warn("Migrating %d stock orderpoint items", cr.rowcount)
+    orderpoints = cr.dictfetchall()
+    # Gather uom data and store it in local caches for later conversions
+    cr.execute('SELECT * from product_uom')
+    uom_data = {}
+    for uom in cr.dictfetchall():
+        uom_data[uom['id']] = uom
+    # Migrate data
+    for op in orderpoints:
+        p_uom_id = op['uom']
+        m_uom_id = op['move_uom']
+        # Check whether uom's are in the same category
+        assert uom_data[p_uom_id]['category_id'] ==\
+            uom_data[m_uom_id]['category_id']
+        factor = 1.0 / uom_data[m_uom_id]['factor'] *\
+            uom_data[p_uom_id]['factor']
+        _min = op['product_min_qty'] * factor
+        _max = op['product_max_qty'] * factor
+        _mul = op['qty_multiple'] * factor
+        logger.warn("""Converting move qty data from %s to %s, factor: %s""",
+                    uom_data[m_uom_id]['name'],
+                    uom_data[p_uom_id]['name'],
+                    factor)
+        Q = '''UPDATE stock_warehouse_orderpoint SET
+                product_min_qty = %s,
+                product_max_qty = %s,
+                qty_multiple = %s
+               WHERE id = %s'''
+        cr.execute(Q, (_min, _max, _mul, op['id']))
+        assert cr.rowcount == 1
+
+
+def migrate_product_supply_method(cr, registry):
+    """Procurements of products: change the supply_method for the matching route
+    make to stock -> MTS Rule: by default
+    make to order -> MTO Rule
+    :param cr:
+    """
+    route_obj = registry['stock.location.route']
+    template_obj = registry['product.template']
+
+    mto_route_id = route_obj.search(
+        cr, uid, [('name', 'like', 'Make To Order')])
+    mto_route_id = mto_route_id and mto_route_id[0] or False
+
+    procure_method_legacy = openupgrade.get_legacy_name('procure_method')
+
+    if mto_route_id:
+        product_ids = []
+        cr.execute("SELECT id FROM product_template WHERE %s = %%s" % (
+            procure_method_legacy,), ('make_to_order',))
+        for res in cr.fetchall():
+            product_ids.append(res[0])
+
+        template_obj.write(
+            cr, uid, product_ids, {'route_ids': [(6, 0, [mto_route_id])]})
+
+
+def migrate_procurement_order(cr, registry):
+    """Set warehouse, partner from the move."""
+    cr.execute(
+        """
+        UPDATE procurement_order AS po
+        SET partner_dest_id = sm.partner_id
+        FROM stock_move AS sm
+        WHERE po.move_dest_id = sm.id
+        """)
+    company_obj = registry['res.company']
+    warehouse_obj = registry['stock.warehouse']
+    procurement_obj = registry['stock.warehouse']
```

Comment:
:question: I guess it should be procurement_obj = registry['procurement.order'] ?
My current log (#l685), indicate :
stock.warehouse.write() with unknown fields: warehouse_id